cf-keycrypt/cf-keycrypt.c

```diff
@@ -403,73 +433,153 @@ static bool CheckHeader(const char *key, const char *value)
             return true;
         }
     }
+    else if (StringSafeEqual(key, "Encrypted-for"))
+    {
+        /* TODO: do some verification that 'value' is valid hash digest? */
+        return true;
+    }
     else
     {
         Log(LOG_LEVEL_ERR, "Unsupported header: '%s'", key);
         return false;
     }
 }
 
-static bool ParseHeaders(FILE *input_file)
+/**
+ * Read from #input_file into #buffer at most #max_chars or until #delimiter is
+ * encountered. If #stop_on_space, also stop when ' ' is read. #buffer is
+ * NUL-terminated when this function returns.
+ *
+ * @warning #buffer has to be at least #max_chars+1 big to accomodate for the
+ *          terminating NUL byte.
+ */
+static inline bool ReadUntilDelim(FILE *input_file, char *buffer, size_t max_chars, char delimiter, bool stop_on_space)
 {
+    bool done = false;
+    size_t i = 0;
+    char c = fgetc(input_file);
+    while (!done && (i <= max_chars))
+    {
+        if (c == EOF)
+        {
+            done = true;
+        }
+        else if (c == delimiter)
+        {
+            done = true;
+            ungetc(c, input_file);
+        }
+        else if (stop_on_space && (c == ' '))
+        {
+            done = true;
+            ungetc(c, input_file);
+        }
+        else
+        {
+            buffer[i] = c;
+            i++;
+            c = fgetc(input_file);
```

Comment:
What happens to this character if `max_chars` is reached?

Reply:
Needs `if (i < max_chars)`. :+1: